File: docs/public/specs/python/area-sine.py
import vgplot as vg

wave = vg.csv("data/m4-area-sine.csv")

brush = vg.selection.intersect()

view = vg.vconcat(
    vg.plot(
        vg.area_y(data="wave", filter_by=brush, x="time_stamp", y="power"),
        vg.y_domain("Fixed"),
        vg.color_domain("Fixed"),
        vg.x_label(None),
        vg.width(680),
        vg.height(180),
    ),
    vg.vspace(5),
    vg.plot(
        vg.area_y(
            data="wave", filter_by=brush, optimize=False, x="time_stamp", y="power"
        ),
        vg.y_domain("Fixed"),
        vg.color_domain("Fixed"),
        vg.x_label(None),
        vg.width(680),
        vg.height(180),
    ),
    vg.vspace(10),
    vg.plot(
        vg.area_y(data="wave", optimize=False, x="time_stamp", y="power"),
        vg.interval_x(bind=brush),
        vg.y_domain("Fixed"),
        vg.width(680),
        vg.height(90),
    ),
)

spec = vg.spec()

File: docs/public/specs/esm/area-sine.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec([
  vg.loadCSV("wave", "data/m4-area-sine.csv")
]);

const $brush = vg.Selection.intersect();

export default vg.vconcat(
  vg.plot(
    vg.areaY(
      vg.from("wave", {filterBy: $brush}),
      {x: "time_stamp", y: "power"}
    ),
    vg.yDomain(vg.Fixed),
    vg.colorDomain(vg.Fixed),
    vg.xLabel(null),
    vg.width(680),
    vg.height(180)
  ),
  vg.vspace(5),
  vg.plot(
    vg.areaY(
      vg.from("wave", {filterBy: $brush, optimize: false}),
      {x: "time_stamp", y: "power"}
    ),
    vg.yDomain(vg.Fixed),
    vg.colorDomain(vg.Fixed),
    vg.xLabel(null),
    vg.width(680),
    vg.height(180)
  ),
  vg.vspace(10),
  vg.plot(
    vg.areaY(
      vg.from("wave", {optimize: false}),
      {x: "time_stamp", y: "power"}
    ),
    vg.intervalX({as: $brush}),
    vg.yDomain(vg.Fixed),
    vg.width(680),
    vg.height(90)
  )
);